src/gridplots.jl

```diff
@@ -103,40 +90,56 @@ function plot_CA1D(data;
     end
   end
 
-  plot2D(node_coords, nodecolors, savename=savename, saveloc=saveloc, saveformat=saveformat, nodesize=nodesize, markeralpha=markeralpha)
+  plot2D(node_coords, nodecolors, markeralpha=markeralpha)
 end
 
 """
     plot_CA2D(data; kwargs...)
 
-Visualizes data of a 2D cellular automaton and saves them to files. Optionally provide a location for the plots to be saved using the `saveloc` argument. The default behavior is to save them in the currently active directory.
+Visualizes data of a 2D cellular automaton.
 
-* `data` output of `CA2D.ca_run`.
+* `data`: output of `CA2D.ca_run`.
 
 # Keywords
 
-* savename::AbstractString = "CA_2D" Plot name
-* saveloc::AbstractString = "./" Optionally provide a location for the plots to be saved
-* saveformat::String="png"
+* t::Int=-1 : The time step to be plotted. If -1, all the rows in `data` are used.
```

Comment:
I think it is more reasonable to use a different type. You can safely use `nothing` instead of `-1`, which will be less prone to error from the user side.

Reply:
Wouldn't it be type unstable if I use `nothing`?

Reply:
No, because `t` is an input argument. Julia creates different versions of each function for different input types. You only have to change the line `if t > -1` to `if t == nothing`.

Reply:
(more importantly, it doesn't matter at all. This simple `if` statement involving `t > -1` has almost no impact on performance)